mutex_protected.h

```diff
@@ -237,8 +278,8 @@ class mutex_protected {
   T v;
 
   // Used by `xyz::lock` when locking multiple mutex_protected objects.
-  mutex_locked<T, std::lock_guard<M>> adopt_lock() {
-    return mutex_locked<T, std::lock_guard<M>>(&v, mutex, std::adopt_lock);
+  mutex_locked<T, moveable_lock_guard<M>> adopt_lock() {
```

Comment:
https://en.cppreference.com/w/cpp/thread/unique_lock/unique_lock is movable, right? Why do we need a custom moveable lock guard?